app/uk/gov/hmrc/overseaspensiontransferbackend/controllers/GetAllSubmissionsController.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.overseaspensiontransferbackend.controllers
+
+import play.api.libs.json.Json
+import play.api.mvc.{AbstractController, Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.overseaspensiontransferbackend.models.PstrNumber
+import uk.gov.hmrc.overseaspensiontransferbackend.models.dtos.GetAllSubmissionsDTO
+import uk.gov.hmrc.overseaspensiontransferbackend.models.submission.SubmissionGetAllResponse
+import uk.gov.hmrc.overseaspensiontransferbackend.services.SubmissionService
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+
+import java.time.Clock
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class GetAllSubmissionsController @Inject() (
+    cc: ControllerComponents,
+    submissionService: SubmissionService,
+    clock: Clock
+  )(implicit ec: ExecutionContext
+  ) extends AbstractController(cc) {
+
+  def getAllSubmissions(pstrNumber: String): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
We should probably add the checkAssociation action to our backend controllers for safety, do you know if thats covered in a separate ticket?

Reply:
I don't believe it is, I don't remember it being mentioned anywhere